apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/AddActionModal.tsx

```diff
@@ -0,0 +1,463 @@
+    },
+    [[], [], []] as [TActionClass[], TActionClass[], TActionClass[]]
+  );
+
+  const handleActionClick = (action: TActionClass) => {
+    setLocalSurvey((prev) => ({
+      ...prev,
+      triggers: prev.triggers.concat(action),
+    }));
+    setOpen(false);
+  };
+
+  return (
+    <div>
+      <Input
+        type="text"
+        onChange={(e) => {
+          setFilteredActionClasses(
+            actionClasses.filter((actionClass) =>
+              actionClass.name.toLowerCase().includes(e.target.value.toLowerCase())
+            )
+          );
+        }}
+        className="mb-2 bg-white"
+        placeholder="Search actions"
+        id="search-actions"
+      />
+      <div className="max-h-96 overflow-y-auto">
+        {[automatic, noCode, code].map(
+          (actions, i) =>
+            actions.length > 0 && (
+              <div key={i}>
+                <h2 className="mb-2 mt-4 font-semibold">
+                  {i === 0 ? "Automatic" : i === 1 ? "No code" : "Code"}
+                </h2>
+                <div className="flex flex-col gap-2">
+                  {actions.map((action) => (
+                    <div
+                      key={action.id}
+                      className="cursor-pointer rounded-md border border-slate-300 bg-white px-4 py-2 hover:bg-slate-100"
+                      onClick={() => handleActionClick(action)}>
+                      <div className="mt-1 flex items-center">
+                        <div className="mr-1.5 h-4 w-4 text-slate-600">
+                          {action.type === "code" ? (
+                            <Code2Icon className="h-4 w-4" />
+                          ) : action.type === "noCode" ? (
+                            <MousePointerClickIcon className="h-4 w-4" />
+                          ) : action.type === "automatic" ? (
+                            <SparklesIcon className="h-4 w-4" />
+                          ) : null}
+                        </div>
+
+                        <h4 className="text-sm font-semibold text-slate-600">{action.name}</h4>
+                      </div>
+                      <p className="mt-1 text-xs text-gray-500">{action.description}</p>
+                    </div>
+                  ))}
+                </div>
+              </div>
+            )
+        )}
+      </div>
+    </div>
+  );
+};
+
+function isValidCssSelector(selector?: string) {
+  if (!selector || selector.length === 0) {
+    return false;
+  }
+  const element = document.createElement("div");
+  try {
+    element.querySelector(selector);
+  } catch (err) {
+    return false;
+  }
+  return true;
+}
+
+interface CreateNewActionProps {
+  actionClasses: TActionClass[];
+  setActionClasses?: React.Dispatch<React.SetStateAction<TActionClass[]>>;
+  isViewer: boolean;
+  setLocalSurvey: React.Dispatch<React.SetStateAction<TSurvey>>;
+  setOpen: React.Dispatch<React.SetStateAction<boolean>>;
+  environmentId: string;
+}
+
+const CreateNewAction = ({
```

Comment:
please use only one component per file to keep files shorter and easier to keep an overview. please separate into multiple files.